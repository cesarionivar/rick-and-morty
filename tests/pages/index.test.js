import React from 'react';
import { render, screen } from '@testing-library/react';
import Home, { getStaticProps } from '../../pages/index';
import { resultApi } from '../mocks/resultApi';

describe('Test in Home page', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  it('should call rick and morty api', async () => {
    fetch.mockResponseOnce(JSON.stringify(resultApi));

    const response = await getStaticProps();
    expect(response).toEqual(
      expect.objectContaining({
        props: {
          characters: resultApi,
        },
      })
    );
  });

  it('renders the title of the page', () => {
    render(<Home characters={resultApi} />);
    expect(
      screen.getByText('The Rick And Morty Characters')
    ).toBeInTheDocument();

    expect(screen.getByText('The Rick And Morty Characters')).toMatchSnapshot();
  });
});
