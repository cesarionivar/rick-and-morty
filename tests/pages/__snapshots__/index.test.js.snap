// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Home page renders the title of the page 1`] = `
<h1>
  The Rick And Morty Characters
</h1>
`;
